progenitor-impl/src/lib.rs

```diff
@@ -262,6 +266,43 @@ impl Generator {
 
         let types = self.type_space.to_stream();
 
+        let extra_impl = TokenStream::from_iter(
+            self.forms
+                .iter()
+                .map(|type_id| {
+                    let typ = self.get_type_space().get_type(type_id).unwrap();
+                    let form_name = typ.name();
+                    let td = typ.details();
+                    let TypeDetails::Struct(tstru) = td else { unreachable!() };
+                    let properties = indexmap::IndexMap::<&'_ str, _>::from_iter(
+                        tstru
+                            .properties()
+                            .filter_map(|(prop_name, prop_id)| {
+                                self.get_type_space()
+                                    .get_type(&prop_id).ok()
+                                    .map(|prop_typ| (prop_name, prop_typ))
+                            })
+                        );
+                    let properties = syn::punctuated::Punctuated::<_, syn::Token![,]>::from_iter(
```

Comment:
```suggestion
                    let properties =
```

#(#properties,)* below

Reply:
The above creates a punctuated list, which can both be used with quote to create the correct expansion, while also being iterable, can change to `Vec<TokenStream>` and use `quote!{ #(#propeties),* if desired.